cypress/components/search/search.cy.tsx

```diff
@@ -328,73 +313,74 @@ context("Test for Search component", () => {
   });
 
   describe("check events for Search component", () => {
-    let callback;
-
-    beforeEach(() => {
-      callback = cy.stub();
-    });
 
     it("should call onClick callback when a click event is triggered", () => {
+      const callback: SearchProps["onClick"] = cy.stub().as("onClick");
+
       CypressMountWithProviders(
         <Search onClick={callback} defaultValue={testCypress} searchButton />
       );
 
-      searchButton()
-        .click()
-        .then(() => {
-          // eslint-disable-next-line no-unused-expressions
-          expect(callback).to.have.been.calledOnce;
-        });
+      searchButton().click();
+      cy.get("@onClick").should("have.been.calledOnce");
+      
     });
 
     it("should call onChange callback when a type event is triggered", () => {
+      const callback: SearchProps["onChange"] = cy.stub().as("onChange");
+
       CypressMountWithProviders(<SearchComponent onChange={callback} />);
 
-      searchDefaultInput()
-        .type("1")
-        .then(() => {
-          // eslint-disable-next-line no-unused-expressions
-          expect(callback).to.have.been.calledOnce;
-        });
+      searchDefaultInput().type("1");
+      cy.get("@onChange").should("have.been.calledOnce");
+      
     });
 
     it("should call onFocus callback when a focus event is triggered", () => {
+      const callback: SearchProps["onFocus"] = cy.stub().as("onFocus");
+
       CypressMountWithProviders(<SearchComponent onFocus={callback} />);
 
-      searchDefaultInput()
-        .focus()
-        .then(() => {
-          // eslint-disable-next-line no-unused-expressions
-          expect(callback).to.have.been.calledOnce;
-        });
+      searchDefaultInput().focus();
+      cy.get("@onFocus").should("have.been.calledOnce");  
     });
 
     it("should call onBlur callback when a blur event is triggered", () => {
+      const callback: SearchProps["onBlur"] = cy.stub().as("onBlur");
+
       CypressMountWithProviders(<SearchComponent onBlur={callback} />);
 
-      searchDefaultInput()
-        .focus()
-        .blur()
-        .then(() => {
-          // eslint-disable-next-line no-unused-expressions
-          expect(callback).to.have.been.calledOnce;
-        });
+      searchDefaultInput().focus().blur();
+      cy.get("@onBlur").should("have.been.calledOnce");    
     });
 
-    it.each([["Enter"], ["Space"]])(
+    it.each([keysToTrigger[0], keysToTrigger[1]])(
```

Comment:
suggestion: You could just leave these `Enter` and `Space` but just change the syntax to `it.each(["Enter", "Space"])(`. They're only used here and not used multiple times else where in the file. It means you can remove the `const` at the start of the file too. Unless @DlgSHi wants these specifically doing it as you've done here?

Reply:
Hi @DipperTheDan , I think this also ok for us :)